tmt/cli.py

```diff
@@ -163,23 +166,32 @@ def main(
         raise SystemExit(0)
 
     # Disable coloring if NO_COLOR is set
-    if 'NO_COLOR' in os.environ:
-        click_contex.color = False
+    # TODO: wouldn't it be nice to support --color/--no-color options?
+    apply_colors = sys.stderr.isatty() and 'NO_COLOR' not in os.environ
+
+    logger = tmt.log.Logger.create(**kwargs)
+    logger.add_console_handler(apply_colors=apply_colors)
+
+    # Propagate color setting to Click as well.
+    click_contex.color = apply_colors
 
     # Save click context and fmf context for future use
     tmt.utils.Common._save_context(click_contex)
 
     # Initialize metadata tree (from given path or current directory)
-    tree = tmt.Tree(path=root or os.curdir)
+    tree = tmt.Tree(logger=logger, path=root or os.curdir)
 
     # TODO: context object details need checks
     click_contex.obj = ContextObject(
-        common=tmt.utils.Common(),
-        fmf_context=tmt.utils.context_to_dict(context),
+        logger=logger,
+        common=tmt.utils.Common(logger=logger),
+        fmf_context=tmt.utils.context_to_dict(logger, context),
         steps=set(),
         tree=tree
         )
 
+    # print('tmt', click_contex.obj.logger)
```

Comment:
Debug line?

Reply:
Fixed in 39891de6dbf9fb4e01471bda84147df2156d9eea